### PR TITLE
Update Keras datasets file handling

### PR DESCRIPTION
**Pull request by VikramTiwari**
using `with` to be consistent with other code in datasets

might also be useful in case file is not present due to corruption of something else

### CODE REVIEW

File: tensorflow/python/keras/datasets/mnist.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def load_data(path='mnist.npz'):
...
       file_hash='8a61469f7ea1b51cbae51d4f78837e45')
-  f = np.load(path)
-  x_train, y_train = f['x_train'], f['y_train']
-  x_test, y_test = f['x_test'], f['y_test']
-  f.close()
+  with np.load(path) as f:
+    x_train, y_train = f['x_train'], f['y_train']
+    x_test, y_test = f['x_test'], f['y_test']
+    
   return (x_train, y_train), (x_test, y_test)
```
Comment (qlzh727) [inline tensorflow/python/keras/datasets/mnist.py]: You can include the return statement in the with context as well.

**Comment by qlzh727** [comment]
@fchollet, I think it worthwhile to clean up the other dataset file too. Using "with" context is always recommended over manual closing.

**Comment by VikramTiwari** [comment]
Hey @qlzh727 I will do that. I can clean up other datasets as well.

File: tensorflow/python/keras/datasets/mnist.py (hunk 1 above)
Comment (VikramTiwari) [reply tensorflow/python/keras/datasets/mnist.py]: Fixed. I have also updated `boston_housing` file.

**Comment by VikramTiwari** [comment]
Ah! I missed that. Fixed now. Thanks for catching it.

**Review by qlzh727 — APPROVED** [review]

**Pull request merged**